File: Algorithms/C++/Kadane_Algo.cpp
#include <iostream>
#include <vector>
using namespace std;

int maxSubArraySum(vector<int> nums)
{
  int maxSum = INT_MIN, currMax = 0;

  for (int i = 0; i < nums.size(); i++)
  {
    currMax = currMax + nums[i];
    if (maxSum < currMax)
      maxSum = currMax;

    if (currMax < 0)
      currMax = 0;
  }
  return maxSum;
}

File: Algorithms/PHP/kadanealgo.php
<?php  
function maxSubArraySum($a, $n) 
{ 
$max_so_far = 0; 
$max_end_here = 0; 
for ($i = 0; $i < $n; $i++) 
{ 
    $max_end_here = $max_end_here + $a[$i]; 
    if ($max_end_here < 0) 
        $max_end_here = 0; 
  
    else if ($max_so_far < $max_end_here) 
        $max_so_far = $max_end_here; 
} 
return $max_so_far; 
}
  
$a = array(1, 1, 2, -1,3, 4, 5); 
$n = count($a); 
$max_sum = maxSubArraySum($a, $n); 
echo "Maximum contiguous sum is ",$max_sum; 
?>